lib/chef/win32/api/net.rb

```diff
@@ -67,6 +83,52 @@ class USER_INFO_3 < FFI::Struct
             :usri3_profile, :LPWSTR,
             :usri3_home_dir_drive, :LPWSTR,
             :usri3_password_expired, :DWORD
+
+          def set(key, val)
+            val = if val.is_a? String
+              encoded = if val.encoding == Encoding::UTF_16LE
+                val
+              else
+                val.to_wstring
+              end
+              val = FFI::MemoryPointer.from_string(encoded)
```

Comment:
If you're setting val =  directly here, you don't need the val = on line 88 or the else block below.